src/task_processor.hpp

```diff
@@ -65,7 +69,10 @@ class TaskProcessor {
   TaskProcessor &operator=(const TaskProcessor &&) = delete;
 
   // Ensure that all tasks are done before destructor returns.
-  ~TaskProcessor() { Wait(); }
+  ~TaskProcessor() { 
+    // If threads were not rejoined earlier, do it now.
+    if ( threads_rejoined_  == false) { Wait(); }
```

Comment:
Did you clang format this? 🧐

Reply:
Whoops, missed a file.  So I can just run `make format` for this right?

Reply:
yep